src/tests/functional/plugin/cpu/single_layer_tests/gather_tree.cpp

```diff
@@ -0,0 +1,173 @@
+
+        init_input_shapes({inputShape, parentShape, maxSeqLenShape, {{}, {{}}}});
+
+        auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+
+        std::shared_ptr<ngraph::Node> inp2;
+        std::shared_ptr<ngraph::Node> inp3;
+        std::shared_ptr<ngraph::Node> inp4;
+
+        auto paramsIn = ngraph::builder::makeDynamicParams(ngPrc, {inputDynamicShapes[0]});
+        if (ngraph::helpers::InputLayerType::PARAMETER == secondaryInputType) {
+            inp2 = ngraph::builder::makeDynamicInputLayer(ngPrc, secondaryInputType, inputDynamicShapes[1]);
+            inp3 = ngraph::builder::makeDynamicInputLayer(ngPrc, secondaryInputType, inputDynamicShapes[2]);
+            inp4 = ngraph::builder::makeDynamicInputLayer(ngPrc, secondaryInputType, inputDynamicShapes[3]);
+
+            paramsIn.push_back(std::dynamic_pointer_cast<ngraph::opset1::Parameter>(inp2));
+            paramsIn.push_back(std::dynamic_pointer_cast<ngraph::opset1::Parameter>(inp3));
+            paramsIn.push_back(std::dynamic_pointer_cast<ngraph::opset1::Parameter>(inp4));
+        } else if (ngraph::helpers::InputLayerType::CONSTANT == secondaryInputType) {
+            auto maxBeamIndex = inputShape.second.front().at(2) - 1;
+
+            inp2 = ngraph::builder::makeConstant<float>(ngPrc, inputShape.second.front(), {}, true, maxBeamIndex);
+            inp3 = ngraph::builder::makeConstant<float>(ngPrc, {inputShape.second.front().at(1)}, {}, true, maxBeamIndex);
+            inp4 = ngraph::builder::makeConstant<float>(ngPrc, {}, {}, true, maxBeamIndex);
+        } else {
+            throw std::runtime_error("Unsupported inputType");
+        }
+
+        auto operationResult = std::make_shared<ngraph::opset4::GatherTree>(paramsIn.front(), inp2, inp3, inp4);
+
+        ngraph::ResultVector results{std::make_shared<ngraph::opset4::Result>(operationResult)};
+        function = std::make_shared<ngraph::Function>(results, paramsIn, "GatherTree");
+    }
+
+    void generate_inputs(const std::vector<ngraph::Shape>& targetInputStaticShapes) override {
+        inputs.clear();
+        const auto maxBeamIndex = targetInputStaticShapes.front().at(2) - 1;
+        const auto& funcInputs = function->inputs();
+
+        for (size_t i = 0; i < funcInputs.size(); ++i) {
+            auto tensor =
+                ov::test::utils::create_and_fill_tensor(funcInputs[i].get_element_type(),
+                                                        targetInputStaticShapes[i],
+                                                        maxBeamIndex,
+                                                        (i == 2 || i == 3) ? maxBeamIndex / 2 : 0);
+            inputs.insert({funcInputs[i].get_node_shared_ptr(), tensor});
+        }
+    }
+};
+
+TEST_P(GatherTreeLayerCPUTest, CompareWithRefs) {
+    run();
+}
+
+namespace {
+
+const std::vector<InferenceEngine::Precision> netPrecisions = {
```

Comment:
Could we use `ElementType` instead of `InferenceEngine::Precision` ? In that way, we could avoid convert inside the `SetUp`

Reply:
done